postgresql/connection.go

```diff
@@ -0,0 +1,82 @@
+package postgresql
+
+import (
+	"database/sql"
+	"fmt"
+	"os"
+
+	"github.com/joho/godotenv"
+
+	_ "github.com/GoogleCloudPlatform/cloudsql-proxy/proxy/dialers/postgres" // blank import
+)
+
+type StatsService struct {
+	db *sql.DB
+}
+
+func NewStatsService(config Config) (StatsService, error) {
+	dbURI := fmt.Sprintf("host=%s user=%s password=%s dbname=%s sslmode=disable",
+		config.Host,
+		config.User,
+		config.Password,
+		config.DBName)
+
+	db, err := sql.Open("cloudsqlpostgres", dbURI)
+	if err != nil {
+		return StatsService{}, ErrDatabaseConnection
+	}
+
+	err = db.Ping()
+	if err != nil {
+		return StatsService{}, ErrDatabasePing
+	}
+
+	db.SetMaxIdleConns(config.IdleConn)
+	db.SetMaxOpenConns(config.MaxConn)
+
+	return StatsService{db}, nil
+}
+
+type Config struct {
+	Host     string
+	User     string
+	Password string
+	DBName   string
+	IdleConn int
+	MaxConn  int
+}
+
+func GetConfigFromEnvVariables() (Config, error) {
+	var config Config
+
+	err := godotenv.Load(".env")
+	// no error returned here because .env is not deployed
+	if err != nil {
+		fmt.Println("no .env file found")
+	}
+
+	config.Host = os.Getenv("PSQL_HOST")
+	if config.Host == "" {
+		return config, ErrEnvironmentVariable
+	}
+
+	config.User = os.Getenv("PSQL_USER")
+	if config.User == "" {
+		return config, ErrEnvironmentVariable
+	}
+
+	config.Password = os.Getenv("PSQL_PASSWORD")
+	if config.Password == "" {
+		return config, ErrEnvironmentVariable
+	}
+
+	config.DBName = os.Getenv("PSQL_NAME")
+	if config.DBName == "" {
+		return config, ErrEnvironmentVariable
+	}
+
+	config.IdleConn = 10
+	config.MaxConn = 25
+
+	return config, nil
+}
```

Comment:
[suggestion] I'd rather do this in the main function:
```go
cfg, err := envConfig() // very early in the function
statsService, err := postgresql.NewStatsService(cfg)

// connection is ok, compute stuff and store to db
```

Reply:
What do you mean by main function? In a main package? There is none at the moment.
I thought we could only connect when the computing of the stats has ended :thinking:

Reply:
Here the "main" function is `worker.Digest`.
Why compute all the data if in the end we cannot connect to the database? That's why I suggest to make all the necessary checks upstream

Reply:
On the opposite, I was thinking why bother connecting to the database if we fail to compute the data? :sweat_smile:

Reply:
I'd say because the computing part is supposed to be heavier and longer task (that's the reason we have a worker after all) 😛 
It's debatable for the connection part I agree, however I'm convinced the params checking at least should be done before anything. Connecting to the db right after getting the config also allows to gather related logics by using the config closer to where it's retrieved.

Reply:
Ok, you've convinced me, let's do it at the beginning of `worker.Digest`since worker should instead be heavier.